src/main/webapp/script.js

```diff
@@ -0,0 +1,371 @@
+ * @param {Image} image
+ * @return {HTMLCanvasElement} canvas DOM element
+ */
+function createCanvasForImage(image) {
+  const canvas = document.createElement('canvas');
+  canvas.setAttribute('width', image.width);
+  canvas.setAttribute('height', image.height);
+
+  return canvas;
+}
+
+/**
+ * Function to handle all the blurring process
+ * and putting both blurred and original images on page.
+ * @param {string} imageUrl
+ */
+async function blur(imageUrl) {
+  // get areas to blur from server
+  const rectsToBlurPromise = getBlurAreas();
+
+  // create Image object from url to put it on canvases
+  const imageObj = new Image();
+  imageObj.src = imageUrl;
+
+  // need to wait until image loads to put it anywhere
+  imageObj.onload = () => {
+    prepareHtmlElements(imageObj);
+
+    // wait for server to respond and finish blurring
+    rectsToBlurPromise.then((rects) => {
+      const blurredImage = getBlurredImage(rects, imageObj);
+      createDownloadButton(blurredImage);
+      showBlurredImage(blurredImage);
+    });
+  };
+}
+
+/**
+ * Function to update html page for blurring a new image.
+ * It resizes canvases and draws new image on them.
+ * It deletes previous download button if there was.
+ * @param {Image} image
+ */
+function prepareHtmlElements(image) {
+  const prepareCanvas = (canvasId) => {
+    const canvas = document.getElementById(canvasId);
+    const ctx = canvas.getContext('2d');
+
+    // clear canvas
+    ctx.clearRect(0, 0, canvas.width, canvas.height);
+
+    // resize canvas height to fit new image
+    canvas.height = image.height * canvas.width / image.width;
+
+    // draw new image on it, scaling the image to fit in canvas
+    ctx.drawImage(image, 0, 0, canvas.width, canvas.height);
+  };
+  prepareCanvas('input-canvas');
+  prepareCanvas('output-canvas');
+
+  // remove download button
+  const downloadButton = document.getElementById('download-button');
+  if (downloadButton !== null) {
+    downloadButton.remove();
+  }
+}
+
+/**
+ * Class for a rectangle to blur.
+ */
+class Rect {
+  /**
+   * @constructor
+   * @param {JsonRect} _rect
+   * Constructs Rect from rectangle object we get from json.
+   * Rectangle object we get from json has 4 points - its corners,
+   * our Rect has top left corner, width and height.
+   */
+  constructor(_rect) {
+    // get minimum of x values from _rect
+    this.leftX = _rect.reduce((min, r) => r.x < min ? r.x : min, _rect[0].x);
+    // get minimum of y values from _rect
+    this.topY = _rect.reduce((min, r) => r.y < min ? r.y : min, _rect[0].y);
+
+    // get maximum of x values from _rect and calculate width as that maximum - minimum
+    const rightX = _rect.reduce((max, r) => r.x > max ? r.x : max, _rect[0].x);
+    this.width = rightX - this.leftX + 1;
+
+    // get maximum of y values from _rect and calculate height as that maximum - minimum
+    const bottomY = _rect.reduce((max, r) => r.y > max ? r.y : max, _rect[0].y);
```

Comment:
Can you get min and max for x and y in one loop? It look like you're doing 4 loops instead.